test/test.jl

```diff
@@ -257,6 +257,18 @@ end
 @test counter_17462_pre == 3
 @test counter_17462_post == 1
 
+# Issue #21008
+ts = try
+    @testset "@test_broken and @test_skip should not give an exception" begin
+        @test_broken false
+        @test_skip true
+        @test_skip false
+    end
+catch
+    nothing # Shouldn't get here
+end
+@test typeof(ts) == Base.Test.DefaultTestSet
```

Comment:
oh btw, `isa` is so much clearer now that it can be used as an infix operator.  Nice move!